### PR TITLE
EVG-7549: add commit queue checkbox

### PR DESCRIPTION
**Pull request by dominoweir**

### CODE REVIEW

File: src/pages/MyPatches.tsx [hunk 1]
```diff
@@ -1,5 +1,27 @@
...
-  return <div>My Patches Page</div>;
+  const { search, pathname } = useLocation();
+  const history = useHistory();
+  const parsed = queryString.parse(search);
+  let showCommitQueue = (parsed.commitQueue === "true");
+  return (
+    <div>
+      <div>My Patches Page</div>
+      <Checkbox
+        onChange={() => {
```
Comment (dominoweir) [inline src/pages/MyPatches.tsx:15]: I'm not sure if this should be in its own separate function, left it inline for now

**Comment by dominoweir** [comment]
My main lingering question on this is how to ensure the box is checked initially, just because I'm not clear on how someone arrives at this page/if the URL will already contain commitQueue=true when they do.

File: src/pages/MyPatches.tsx (hunk 1 above)
Comment (dominoweir) [reply src/pages/MyPatches.tsx]: moved to a function after discovering the linter takes issue with lambdas

**Comment by SupaJoon** [comment]
> My main lingering question on this is how to ensure the box is checked initially, just because I'm not clear on how someone arrives at this page/if the URL will already contain commitQueue=true when they do.

The checkbox would be checked on initial render if the commitQueue query param was included in the URL on initial render. This scenario would happen if you were to copy and paste the URL to someone after clicking the checkbox.

File: src/pages/MyPatches.tsx [hunk 2]
```diff
@@ -1,5 +1,30 @@
...
+  const { search, pathname } = useLocation();
+  const history = useHistory();
+  const parsed = queryString.parse(search);
+  let showCommitQueue = parsed.commitQueue === "true";
+
+  function onCheckboxChange() {
+    showCommitQueue = !showCommitQueue;
+    history.push({
+      pathname: pathname,
+      search: "?commitQueue=" + showCommitQueue
```
Comment (SupaJoon) [inline src/pages/MyPatches.tsx:16]: this line will have to be updated as we add more query params to the page which could make things confusing to maintain later on. Check out this function that passes on existing query params https://github.com/evergreen-ci/spruce/blob/new-ui-feature-branch/src/pages/patch/patchTabs/tasks/TasksTable.tsx#L31

File: src/pages/MyPatches.tsx [hunk 3]
```diff
@@ -1,5 +1,30 @@
...
+import { useLocation, useHistory } from "react-router-dom";
+import queryString from "query-string";
+import Checkbox from "@leafygreen-ui/checkbox";
 
 export const MyPatches = () => {
-  return <div>My Patches Page</div>;
+  const { search, pathname } = useLocation();
+  const history = useHistory();
+  const parsed = queryString.parse(search);
+  let showCommitQueue = parsed.commitQueue === "true";
```
Comment (SupaJoon) [inline src/pages/MyPatches.tsx]: `commitQueue` should be saved inside an interface that defines query params for this page like what we did for Tasks table on the Patch page: https://github.com/evergreen-ci/spruce/blob/new-ui-feature-branch/src/types/task.ts#L10

File: src/pages/MyPatches.tsx [hunk 4]
```diff
@@ -1,5 +1,35 @@
...
+import queryString from "query-string";
+import Checkbox from "@leafygreen-ui/checkbox";
+import { MyPatchesQueryParams } from "types/task";
 
 export const MyPatches = () => {
-  return <div>My Patches Page</div>;
+  const { search, pathname } = useLocation();
+  const { replace } = useHistory();
+  const parsed = queryString.parse(search);
+  let showCommitQueue = parsed.commitQueue
```
Comment (SupaJoon) [inline src/pages/MyPatches.tsx:11]: this can be showCommitQueue = parsed.commitQueue === "true"

File: src/pages/MyPatches.tsx [hunk 5]
```diff
@@ -1,5 +1,35 @@
...
+    ? parsed.commitQueue === "true"
+    : true;
+
+  function onCheckboxChange() {
+    showCommitQueue = !showCommitQueue;
+    replace(
+      `${pathname}?${queryString.stringify({
+        ...queryString.parse(search),
+        [MyPatchesQueryParams.CommitQueue]: showCommitQueue
```
Comment (SupaJoon) [inline src/pages/MyPatches.tsx:20]: and here the value can be !showCommitQueue

File: src/pages/MyPatches.tsx [hunk 6]
```diff
@@ -1,5 +1,35 @@
...
-  return <div>My Patches Page</div>;
+  const { search, pathname } = useLocation();
+  const { replace } = useHistory();
+  const parsed = queryString.parse(search);
+  let showCommitQueue = parsed.commitQueue
+    ? parsed.commitQueue === "true"
+    : true;
+
+  function onCheckboxChange() {
+    showCommitQueue = !showCommitQueue;
```
Comment (SupaJoon) [inline src/pages/MyPatches.tsx:16]: we should avoid reassigning variables when we can

File: src/types/task.ts [hunk 7]
```diff
@@ -17,6 +17,10 @@ export enum PatchTasksQueryParams {
   TaskName = "taskName"
 }
 
+export enum MyPatchesQueryParams {
```
Comment (SupaJoon) [inline src/types/task.ts:20]: we should move enum to `src/types/patch.ts`

File: src/pages/MyPatches.tsx [hunk 8]
```diff
@@ -1,5 +1,32 @@
...
+import { MyPatchesQueryParams } from "types/patch";
 
 export const MyPatches = () => {
-  return <div>My Patches Page</div>;
+  const { search, pathname } = useLocation();
+  const { replace } = useHistory();
+  const parsed = queryString.parse(search);
+  const showCommitQueue = parsed.commitQueue === "true" || parsed.commitQueue === undefined;
+
+  function onCheckboxChange() {
```
Comment (SupaJoon) [inline src/pages/MyPatches.tsx:13]: can we convert this to an arrow function?

**Comment by dominoweir** [comment]
Tests have been added 👍

File: cypress/integration/my_patches.js [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+  });
+
+  it("Should render with Show Commit Queue box checked when commitQueue not indicated in URL query param", () => {
+    cy.visit(MY_PATCHES_ROUTE);
+    cy.get(".cy-checkbox")
+      .find("input")
+      .should("be.checked");
+  });
+
+  it("Should render with Show Commit Queue box checked when commitQueue is false in URL query param", () => {
```
Comment (tgrander) [inline cypress/integration/my_patches.js:18]: i think this should be "Should render with Show Commit Queue box UNchecked when commitQueue is false in URL query param"

File: cypress/integration/my_patches.js [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+  "/my-patches";
+describe("my patches page", function() {
+  beforeEach(() => {
+    cy.server();
+    cy.login();
+  });
+
+  it("Should render with Show Commit Queue box checked when commitQueue not indicated in URL query param", () => {
+    cy.visit(MY_PATCHES_ROUTE);
+    cy.get(".cy-checkbox")
```
Comment (tgrander) [inline cypress/integration/my_patches.js:13]: some of the old selectors will be classes but we have moved to using data-cy attributes on the elements for testing.
```
// JSX
<div data-cy="commit-queue-checkbox">I'm the checkbox element</div>

// test file
cy.get("[data-cy=commit-queue-checkbox]")
```

**Pull request merged**